gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemIntegrationTest.java

```diff
@@ -573,6 +613,39 @@ public void testGlobStatus() throws IOException {
     ghfs.delete(testRoot, true);
   }
 
+  @Test
+  public void testGlobStatusOptions() throws IOException {
+    testGlobStatusFlatConcurrent(true,true);
+    testGlobStatusFlatConcurrent(true,false);
+    testGlobStatusFlatConcurrent(false,true);
+    testGlobStatusFlatConcurrent(false,false);
+  }
+
+  private void testGlobStatusFlatConcurrent(boolean flat, boolean concurent) throws IOException {
+    Configuration configuration = getConfigurationWtihImplementation();
+    configuration.set(GoogleHadoopFileSystemConfiguration.GCS_FLAT_GLOB_ENABLE.getKey(), String.valueOf(flat));
+    configuration.set(GoogleHadoopFileSystemConfiguration.GCS_CONCURRENT_GLOB_ENABLE.getKey(), String.valueOf(concurent));
+
+    ghfs.initialize(ghfs.getUri(),configuration);
+
+    Path testRoot = new Path("/directory1/");
+    ghfs.mkdirs(testRoot);
+    ghfs.mkdirs(new Path("/directory1/subdirectory1"));
+
+    byte[] data = new byte[10];
+    for (int i = 0; i < data.length; i++) {
+      data[i] = (byte) i;
+    }
+
+    createFile(new Path("/directory1/subdirectory1/file1"), data);
```

Comment:
You can just convert string to byte array, instead of creating it manually:
`"data".getBytes(StandardCharsets.UTF_8)`